src/datasets/utils/metadata.py

```diff
@@ -0,0 +1,188 @@
+import json
+import logging
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+
+# loading package files: https://stackoverflow.com/a/20885799
+try:
+    import importlib.resources as pkg_resources
+except ImportError:
+    # Try backported to PY<37 `importlib_resources`.
+    import importlib_resources as pkg_resources
```

Comment:
We should add `importlib_resources` to the test dependencies for python<3.7 then ?

This makes me also think we should have a CI for python 3.6

Reply:
tbh I added this copy pasting from SO, I didn't look at the repo's policy for python versions support.

Reply:
You just need to add
```python
"importlib_resources;python_version<'3.7'"
```
to the metadata validation requirements:
```python
    # metadata validation
    "langcodes[data]>=3.1.0",
    "pydantic>=1.8.1",
    "importlib_resources;python_version<'3.7'",
```

Reply:
ooh i understand now, i thought it was already in the stdlib but under a different name!